src/test/java/com/google/api/codegen/testsrc/common/library.proto

```diff
@@ -453,11 +429,15 @@ message Book {
 
 // A single book in the archives.
 message BookFromArchive {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/BookFromArchive",
+    pattern: "archives/{archive}/books/{book}",
+  };
   // The resource name of the book.
   // Book names have the form `archives/{archive_id}/books/{book_id}`.
   string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = "ArchivedBook"];
+    (google.api.field_behavior) = REQUIRED
```

Comment:
Can we keep the resource annotation on this field? Or is like magically implied because the message type has a resource declaration?

Reply:
Removing this was a mistake, will add back